File: 17.Backend Security - Everything You Need to Know/code/typescript/the_three_critical_cookie_flags_4.ts
import { Response } from "express";

export function setSessionCookie(res: Response, sessionID: string) {
    res.cookie("session_id", sessionID, {
        httpOnly: true,          // JS cannot access
        secure: true,            // HTTPS only
        sameSite: "strict",      // no cross-site
        maxAge: 7 * 24 * 3600 * 1000, // 7 days in ms
        path: "/",
    });
}

File: 17.Backend Security - Everything You Need to Know/code/rust/the_three_critical_cookie_flags_4.rs
use actix_web::cookie::{Cookie, SameSite};
use time::Duration;

pub fn set_session_cookie(session_id: &str) -> Cookie<'static> {
    Cookie::build("session_id", session_id.to_string())
        .http_only(true)           // JS cannot access
        .secure(true)              // HTTPS only
        .same_site(SameSite::Strict) // no cross-site
        .max_age(Duration::days(7)) // 7 days
        .path("/")
        .finish()
}
